### PR TITLE
Fixed #27225 -- Made FetchFromCacheMiddleware set Age header for cached responses.

### PR DESCRIPTION
**Pull request by renskiy**
https://code.djangoproject.com/ticket/27225

### CODE REVIEW

File: tests/cache/tests.py [hunk 1]
```diff
@@ -2162,6 +2162,31 @@ def test_304_response_has_http_caching_headers_but_not_cached(self):
...
+    def test_cache_control_max_age(self):
+        view = cache_page(2)(hello_world_view)
+        request = self.factory.get('/view/')
+
+        # Request once
+        with mock.patch.object(time, 'time', return_value=1468749600):  # Sun, 17 Jul 2016 10:00:00 GMT
+            response = view(request)
+            response.close()
+            self.assertIn('Expires', response)
+            self.assertEqual('Sun, 17 Jul 2016 10:00:02 GMT', response['Expires'])
```
Comment (timgraham) [inline tests/cache/tests.py]: Please swap the order in the assertions and put `response` on the left, `self.assertEqual(response['Expires'], 'Sun, 17 Jul 2016 10:00:02 GMT')`

File: django/utils/cache.py [hunk 2]
```diff
@@ -72,8 +72,10 @@ def dictvalue(t):
     # If there's already a max-age header but we're being asked to set a new
     # max-age, use the minimum of the two ages. In practice this happens when
     # a decorator and a piece of middleware both operate on a given view.
-    if 'max-age' in cc and 'max_age' in kwargs:
-        kwargs['max_age'] = min(int(cc['max-age']), kwargs['max_age'])
+    if 'max_age' in kwargs:
```
Comment (timgraham) [inline django/utils/cache.py]: Could you unit test these changes in `test_patch_cache_control`? It's not obvious to me the reason for the changes. Is any update to the comment needed?

File: django/middleware/cache.py [hunk 3]
```diff
@@ -144,6 +147,14 @@ def process_request(self, request):
             request._cache_update_cache = True
             return None  # No cache information available, need to rebuild.
 
+        # Setting cache Age
```
Comment (timgraham) [inline django/middleware/cache.py]: Could you expand the comment to describe the calculation a little bit more? The logic isn't obvious to me.

File: tests/cache/tests.py [hunk 4]
```diff
@@ -2162,6 +2162,31 @@ def test_304_response_has_http_caching_headers_but_not_cached(self):
...
+            self.assertIn('Cache-Control', response)
+            self.assertEqual('max-age=2', response['Cache-Control'])
+            self.assertNotIn('Age', response)
+
+        # Response for second request made later must have Age header
```
Comment (timgraham) [inline tests/cache/tests.py]: Could you add comments that describe the calculations that lead the expected Age value?

**Comment by timgraham** [comment]
Are you addressing the other comments later or did you miss them?

**Comment by renskiy** [comment]
Later this week

**Comment by renskiy** [comment]
Need additional couple of weeks to complete. Sorry for this, no time - preparing to conference.

**Comment by timgraham** [comment]
@renskiy, do you have time to finish this soon?

**Comment by renskiy** [comment]
@timgraham, Yes I have. This weekend I think.

**Comment by LuRsT** [comment]
A super gentle reminder @renskiy. Do you need help?

**Comment by felixxm** [comment]
Closing due to inactivity.

**Pull request closed without merging**